### PR TITLE
chore: pre-commit secrets scanner

### PR DESCRIPTION
**Pull request by vuhuucuong**
Add pre-commit hook to scan for secrets before commit
### FLOW:
1. GET AWS secrets from environment variables and `~/.aws/credentials`
2. Merge those secrets into one file (temporary file)
3. Register AWS provider for git-secrets using the file in step 2
4. Scan for secrets in staged files using `git secrets --scan`
5. Remove temporary file in step 2

### CODE REVIEW

File: scripts/utils/git-secrets.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    if (stderrAddProvider) throw stderrAddProvider
+    console.log('Added AWS provider\n', stdoutAddProvider)
+
+    // scan for secrets
+    const { error: errorScan, stdout: stdoutScan, stderr: stderrScan } = shell.exec('git secrets --scan --cached', {
+      stdio: 'inherit',
+    })
+    if (errorScan) throw errorScan
+    if (stderrScan) throw stderrScan
+    console.log('No secrets found in staged files!\n', stdoutScan)
```
Comment (vuhuucuong) [inline scripts/utils/git-secrets.js:65]: I tried to chain these two commands using &&, but it doesn't work as expected, need to separate into two commands

File: scripts/utils/git-secrets.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    const { error: errorScan, stdout: stdoutScan, stderr: stderrScan } = shell.exec('git secrets --scan --cached', {
+      stdio: 'inherit',
+    })
+    if (errorScan) throw errorScan
+    if (stderrScan) throw stderrScan
+    console.log('No secret found in staged files!\n', stdoutScan)
+    removeTempCredFile()
+  } catch (err) {
+    console.error(err)
+    removeTempCredFile()
```
Comment (phmngocnghia) [inline scripts/utils/git-secrets.js:69]: I thinks this part could result unhandled error if `unlinksync` is fail to execute and throws error

LGTM. Nice 👍

**Comment by vuhuucuong** [comment]
> For me i don't use aws credential file, only use bash profile, are there any way to check it ?

@tanphamhaiduong I've already checked both authentication mechanisms.

If you use bash profile, it means you're exporting `AWS_ACCESS_KEY_ID` & `AWS_SECRET_ACCESS_KEY` as environment variables.
I covered it in here https://github.com/reapit/foundations/pull/1170/files#diff-1d82deb06aec9c6a91d1e347cd934da5R11-R27

Just updated comment on that function to avoid confusing

File: scripts/utils/git-secrets.js (hunk 2 above)
Comment (vuhuucuong) [reply scripts/utils/git-secrets.js:71]: Yes if it throws an error, then just let it does, the script will auto exit with an error code.
If all things go well, then no error is returned.

**Pull request merged**